polars/polars-core/src/series/implementations.rs

```diff
@@ -521,8 +521,20 @@ macro_rules! impl_dyn_series {
                 self.0.append_array(other)
             }
 
-            fn slice(&self, offset: usize, length: usize) -> Result<Series> {
-                self.0.slice(offset, length).map(|ca| ca.into_series())
+            fn slice(&self, offset: i64, length: usize) -> Result<Series> {
+                let offset_abs = offset.abs() as usize;
```

Comment:
Could this logic be implemented at the `ChunkedArray` level? Then we have a central place that does all the slicing logic, and they all have the same behavior.

Reply:
I agree with your comment. It might make more sense to do it in ChunkedArray.
I'' have a look at it and see how it goes